components/core/src/clp/streaming_archive/writer/Archive.cpp

```diff
@@ -647,6 +644,19 @@ void Archive::update_metadata() {
     }
 }
 
+auto Archive::update_global_metadata() -> void {
```

Comment:
How about renaming `update_metadata` to `update_local_metadata` so that there's congruency?